### PR TITLE
Staging/adxl313 maxim

### PR DESCRIPTION
**Pull request by danmois**
Add the Maxim platform to the eval-adxl313z project.

Tested with MAX32650.

### CODE REVIEW

File: projects/eval-adxl313z/src/platform/maxim/parameters.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	.flow = UART_FLOW_DIS
+};
+#endif
+
+struct max_spi_init_param xsip  = {
+	.numSlaves = 1,
+	.polarity = SPI_SS_POL_LOW,
+	.vssel = 1 // 3.3V
+};
+
```
Comment (amiclaus) [inline projects/eval-adxl313z/src/platform/maxim/parameters.c:59]: nit: extra line

File: projects/eval-adxl313z/src/platform/maxim/parameters.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#ifdef BASIC_EXAMPLE
+struct max_uart_init_param xuip = {
+	.flow = UART_FLOW_DIS
+};
+#endif
+
+struct max_spi_init_param xsip  = {
+	.numSlaves = 1,
+	.polarity = SPI_SS_POL_LOW,
+	.vssel = 1 // 3.3V
```
Comment (amiclaus) [inline projects/eval-adxl313z/src/platform/maxim/parameters.c:57]: can't we use MXC_GPIO_VSSEL_VDDIOH?

File: projects/eval-adxl313z/src/platform/maxim/main.c [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+		goto error;
+#endif
+
+#if (IIO_EXAMPLE + BASIC_EXAMPLE > 1)
+#error Selected example projects cannot be enabled at the same time. \
+Please enable only one example and re-build the project.
+#endif
+
+error:
+	return ret;
```
Comment (amiclaus) [inline projects/eval-adxl313z/src/platform/maxim/main.c:96]: `no_os_uart_remove`

File: projects/eval-adxl313z/src/platform/maxim/platform_src.mk [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+SRCS += $(PLATFORM_DRIVERS)/maxim_delay.c     \
+        $(PLATFORM_DRIVERS)/maxim_gpio.c      \
+        $(PLATFORM_DRIVERS)/maxim_spi.c       \
+        $(PLATFORM_DRIVERS)/maxim_rtc.c       \
+        $(PLATFORM_DRIVERS)/maxim_gpio_irq.c  \
+        $(PLATFORM_DRIVERS)/maxim_irq.c       \
+        $(PLATFORM_DRIVERS)/maxim_uart.c      \
+        $(PLATFORM_DRIVERS)/maxim_init.c      \
+        $(PLATFORM_DRIVERS)/maxim_uart_stdio.c
+
```
Comment (amiclaus) [inline projects/eval-adxl313z/src/platform/maxim/platform_src.mk:21]: nit: extra line

**Review by amiclaus — COMMENTED** [review]
please rebase onto master branch. 2 of the commits are already there.

**Comment by danmois** [comment]
V2
- rebase

File: projects/eval-adxl313z/src/platform/maxim/parameters.c (hunk 2 above)
Comment (danmois) [reply projects/eval-adxl313z/src/platform/maxim/parameters.c:57]: Yes, we can.

File: projects/eval-adxl313z/src/platform/maxim/main.c (hunk 3 above)
Comment (danmois) [reply projects/eval-adxl313z/src/platform/maxim/main.c:98]: Called if basic_example_main() fails.

**Comment by danmois** [comment]
V3
- remove extra lines
- call no_os_uart_remove when basic_example_main fails
- use MXC_GPIO_VSSEL_VDDIOH define

File: projects/eval-adxl313z/src/platform/maxim/parameters.h [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+#define UART_IRQ_ID     UART0_IRQn
+#endif
+
+#define UART_DEVICE_ID  0
+#define UART_BAUDRATE   57600
+#define UART_EXTRA      &xuip
+#define UART_OPS        &max_uart_ops
+
+#if (TARGET_NUM == 78000)
+#define    1
```
Comment (buha) [inline projects/eval-adxl313z/src/platform/maxim/parameters.h:66]: SPI_DEVICE_ID missing

Comment (danmois) [reply projects/eval-adxl313z/src/platform/maxim/parameters.h]: Added.

**Comment by danmois** [comment]
V4

- add SPI_DEVICE_ID for max78000

**Review by amiclaus — APPROVED** [review]

**Pull request merged**